tests/Cache/FileCacheManagerTest.php

```diff
@@ -351,38 +217,170 @@ public function testSetFileUsesRelativePathToFile(): void
         $relativePathToFile = 'src/hello.php';
         $fileContent = '<?php echo "Hello!"';
 
-        $directoryProphecy = $this->prophesize(\PhpCsFixer\Cache\DirectoryInterface::class);
-        $directoryProphecy->getRelativePathTo(Argument::is($file))->willReturn($relativePathToFile);
-
-        $cachedSignature = $this->prophesize(SignatureInterface::class)->reveal();
-
-        $signatureProphecy = $this->prophesize(SignatureInterface::class);
-        $signatureProphecy->equals(Argument::is($cachedSignature))->willReturn(true);
-        $signature = $signatureProphecy->reveal();
-
-        $cacheProphecy = $this->prophesize(CacheInterface::class);
-        $cacheProphecy->getSignature()->willReturn($cachedSignature);
-        $cacheProphecy->set(Argument::is($relativePathToFile), Argument::is(md5($fileContent)))->shouldBeCalled();
-        $cache = $cacheProphecy->reveal();
+        $directory = $this->getDirectoryDouble($relativePathToFile);
+        $cachedSignature = $this->getSignatureDouble(true);
+        $signature = $this->getSignatureDouble(true);
 
-        $handlerProphecy = $this->prophesize(FileHandlerInterface::class);
-        $handlerProphecy->read()->willReturn($cache);
-        $handlerProphecy->getFile()->willReturn($cacheFile);
-        $handlerProphecy->write(Argument::is($cache));
-        $handler = $handlerProphecy->reveal();
+        $cache = $this->getCacheDouble($cachedSignature);
+        $handler = $this->getFileHandlerDouble($cache, $cacheFile);
 
-        $manager = new FileCacheManager(
-            $handler,
-            $signature,
-            false,
-            $directoryProphecy->reveal()
-        );
+        $manager = new FileCacheManager($handler, $signature, false, $directory);
 
         $manager->setFile($file, $fileContent);
+
+        self::assertTrue($cache->has($relativePathToFile));
+        self::assertSame(md5($fileContent), $cache->get($relativePathToFile));
     }
 
     private function getFile(): string
     {
         return __DIR__.'/../Fixtures/.php_cs.empty-cache';
     }
+
+    private function getDirectoryDouble(string $relativePathToFile): DirectoryInterface
+    {
+        return new class($relativePathToFile) implements DirectoryInterface {
+            private string $relativePathToFile;
+
+            public function __construct(string $relativePathToFile)
+            {
+                $this->relativePathToFile = $relativePathToFile;
+            }
+
+            public function getRelativePathTo(string $file): string
+            {
+                return $this->relativePathToFile;
+            }
+        };
+    }
+
+    private function getSignatureDouble(bool $isEqual): SignatureInterface
```

Comment:
not sure if we can do any better, but with invocations like `$this->getSignatureDouble(true)` it is not intuitive what `true` stands for.
yes, sure, one can go to method declaration.
no good suggestion from my side how to improve

Reply:
when we drop support for PHP 7 we can use named parameters